Assets/HoloToolkit/Input/Scripts/Voice/KeywordManager.cs

```diff
@@ -40,22 +40,27 @@ public enum RecognizerStartBehavior { AutoStart, ManualStart };
         [Tooltip("An enumeration to set whether the recognizer should start on or off.")]
         public RecognizerStartBehavior RecognizerStart;
 
-        [Tooltip("An array of string keywords and UnityEvents, to be set in the Inspector.")]
-        public KeywordAndResponse[] KeywordsAndResponses;
+        [Tooltip("An array of string keywords and keys, to be set in the Inspector.")]
+        public KeywordAndKeyCode[] KeywordsAndKeys;
 
         private KeywordRecognizer keywordRecognizer;
-        private Dictionary<string, UnityEvent> responses;
 
-        void Start()
+        public override SupportedInputEvents SupportedEvents
         {
-            if (KeywordsAndResponses.Length > 0)
+            get
             {
-                // Convert the struct array into a dictionary, with the keywords and the keys and the methods as the values.
-                // This helps easily link the keyword recognized to the UnityEvent to be invoked.
-                responses = KeywordsAndResponses.ToDictionary(keywordAndResponse => keywordAndResponse.Keyword,
-                                                              keywordAndResponse => keywordAndResponse.Response);
+                return SupportedInputEvents.SourceUpAndDown;
```

Comment:
You probably need to add a new type of SupportedInputEvents (Speech) and change this accordingly